### PR TITLE
Fix relative entrypoint on hydra client

### PR DESCRIPTION
**Pull request by ginifizz**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  |no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #124
| License       | MIT
| Doc PR        | api-platform/doc#1234

### CODE REVIEW

File: src/hydra/hydraClient.js [hunk 1]
```diff
@@ -168,8 +168,9 @@ export default ({entrypoint, resources = []}, httpClient = fetchHydra) => {
    * @returns {Object}
    */
   const convertReactAdminRequestToHydraRequest = (type, resource, params) => {
-    const collectionUrl = new URL(`${entrypoint}/${resource}`);
-    const itemUrl = new URL(params.id, entrypoint);
+    const entrypointUrl = new URL(entrypoint, window.location.origin);
```
Comment (dunglas) [inline src/hydra/hydraClient.js]: We could even use `window.location.href` to support relative paths (as in your test)

Comment (arnaudoisel) [reply src/hydra/hydraClient.js]: I don't see any difference between window.location.origin and window.location.href, except that window.location.href would include potential anchors (https://domain#foo), but I think they are treated the same way by the URL class.

File: src/hydra/hydraClient.js [hunk 2]
```diff
@@ -168,8 +168,9 @@ export default ({entrypoint, resources = []}, httpClient = fetchHydra) => {
...
+    const collectionUrl = new URL(`${entrypoint}/${resource}`, entrypointUrl);
```
Comment (arnaudoisel) [inline src/hydra/hydraClient.js]: Could it be :
```suggestion
    const collectionUrl = new URL(resource, entrypointUrl);
```

File: src/hydra/hydraClient.js (hunk 1 above)
Comment (arnaudoisel) [reply src/hydra/hydraClient.js]: I don't see any difference between window.location.origin and window.location.href, except that window.location.href would include potential anchors (https://domain#foo), but I think they are treated the same way by the URL class.

File: src/hydra/hydraClient.js (hunk 2 above)
Comment (arnaudoisel) [inline src/hydra/hydraClient.js]: Could it be :
```suggestion
    const collectionUrl = new URL(resource, entrypointUrl);
```

File: src/hydra/hydraClient.js (hunk 1 above)
Comment (dunglas) [reply src/hydra/hydraClient.js]:
```
> window.location.href
< "https://github.com/api-platform/admin/pull/126"
> window.location.origin
< "https://github.com"
```

With `window.location.href`, relative path will work.

Comment (dunglas) [reply src/hydra/hydraClient.js]:
```
> new URL('foo/bar', window.location.origin)
< URL {href: "https://github.com/foo/bar", origin: "https://github.com", protocol: "https:", username: "", password: "", …}
> new URL('foo/bar', window.location.href)
< URL {href: "https://github.com/api-platform/admin/pull/foo/bar", origin: "https://github.com", protocol: "https:", username: "", password: "", …}
```

Comment (arnaudoisel) [reply src/hydra/hydraClient.js]: Right, sorry, my bad. I was only testing in the case where the base URL is the origin...

```
> new URL('foo/bar', (new URL('https://github.com/api-platform')).origin)
< URL { href: "https://github.com/foo/bar" }
> new URL('foo/bar', (new URL('https://github.com/api-platform')).href)
< URL { href: "https://github.com/foo/bar" }
```

Comment (dunglas) [reply src/hydra/hydraClient.js]: Wow, it looks like a browser bug to me. The origin should never include the path.

Comment (arnaudoisel) [reply src/hydra/hydraClient.js]: It doesn't, my example is just the special case where the path length is 1. In that case the base URL is the origin.

**Pull request closed without merging**